### PR TITLE
Fix external link indicator in Link Control

### PR DESCRIPTION
**Pull request by richtabor**
## What?
Follow up to https://github.com/WordPress/gutenberg/pull/60255 for Link Control. 

## Testing Instructions

1. Open a post or page.
2. Add a link.
3. See changes.

## Screenshots or screencast 

| Before  | After |
| ------------- | ------------- |
|CleanShot 2024-04-03 at 15 39 13|CleanShot 2024-04-03 at 15 38 50|

### CODE REVIEW

**Review by jasmussen — APPROVED** [review]

**Review by t-hamano — CHANGES_REQUESTED** [review]
If the text is long, it seems like it won't be one line after applying this PR.

### trunk

trunk

### This PR

pr

I think this is because the `display: -webkit-box` for truncating the text is overwritten with the `inline` value:

inline-override

Is it possible to adjust the appearance while maintaining the current number of lines?

**Comment by t-hamano** [comment]
From brief testing, the following style appears to solve the problem.

```css
.block-editor-link-control__search-item .block-editor-link-control__search-item-title {
    display: flex;
}
```

**Comment by jasmussen** [comment]
Back in the day, one hack to avoid breaking a word in two despite there being a space, was to insert a non breaking space character, `&nbsp;` between the words. That effectively makes it count as "one word". So I wonder if a fix is to do `[linktitle]&nbsp;↗`?

**Comment by richtabor** [comment]
> From brief testing, the following style appears to solve the problem.

That didn't quite work either, as it renders all the way to the right of the available area, regardless of truncation. 

CleanShot 2024-04-08 at 21 19 45

**Comment by richtabor** [comment]
I realized that trunk intentionally hides the previous external link SVG icon. 

I updated this to target the .components-external-link__icon class, rather than `svg`, to do the same.

**Comment by t-hamano** [comment]
> I realized that trunk intentionally hides the previous external link SVG icon. 
> I updated this to target the .components-external-link__icon class, rather than `svg`, to do the same. 

I also agree with this approach. This is because the link may not be an external resource such as an anchor.

@jasmussen What do you think?

**Comment by jasmussen** [comment]
I.e. hide the external link entirely, as was the case before? Sounds good, there's likely good past context for that good to not accidentally revert.

**Review by t-hamano — APPROVED** [review]

**Comment by t-hamano** [comment]
Sorry, I accidentally sent a re-review to @jasmussen 😅

**Comment by richtabor** [comment]
Rebased to get tests ✅

File: packages/block-editor/src/components/link-control/style.scss [hunk 1]
```diff
@@ -268,7 +268,7 @@ $block-editor-link-control-number-of-actions: 1;
 			font-weight: normal;
 		}
 
-		svg {
+		.components-external-link__icon {
```
Comment (jeryj) [inline packages/block-editor/src/components/link-control/style.scss:272]: This element also has "Opens in a new tab" as an `aria-label` for screen readers to announce that it's an external link, and if it's set to `display: none`, it doesn't get announced. Might need to raise the specificity with `.components-external-link__icon svg`?

Comment (richtabor) [reply packages/block-editor/src/components/link-control/style.scss:272]: There's not an SVG anymore; that's the issue.

CleanShot 2024-04-10 at 17 45 44

This is what we had before https://github.com/WordPress/gutenberg/pull/60255:

CleanShot 2024-04-10 at 17 47 09

Comment (richtabor) [reply packages/block-editor/src/components/link-control/style.scss:272]: @jameskoster any ideas here?

**Pull request merged**